### PR TITLE
Create v1.4.0 release notes

### PR DESCRIPTION
**Pull request by jillian-maroket**
| Section | Status | Notes |
| --- | --- | --- |
| Downloads | Pending | To be provided by engineering |
| Installation | Complete | Reused v1.3.0 content, updated links |
| Upgrades | Complete | Reused v1.3.0 content, updated links |
| Highlights | Complete | No showcase video available for "Third-Party Storage for Diskless Servers" |
| Enhancements | Pending | To be provided by engineering |
| Bug Fixes | Pending | To be provided by engineering |
| Known Issues | Pending | To be provided by engineering |
| Components | Pending | To be provided by engineering |
| Contributors | Pending | Final contributor list TBD |

@asettle @bk201 Please review the content of the **Highlights** section.

Close https://github.com/harvester/release-notes/issues/28

### CODE REVIEW

**Comment by tserong** [comment]
For discussion: the enhancements list I just added tries to put the gui issues with the related main issues, e.g.:

```
- [FEATURE] Host local storage support - Experimental  [#5724](https://github.com/harvester/harvester/issues/5724)
- [GUI] [FEATURE] Host local storage support [#6057](https://github.com/harvester/harvester/issues/6057)
```

But it kinda looks a bit weird to me - the GUI issue is technically redundant given it's usually linked off the main issue anyway.

OTOH, we have GUI issues that don't _have_ a main issue - see the set at the end of the list.

Maybe it makes more sense to group the main issues and gui issues separately by type rather than trying to tie them together?

Thoughts @jillian-maroket @bk201 ?

Also the order is somewhat arbitrary -- features first then enhancements, but beyond that it's whatever order the release tool listed the issues in

**Comment by tserong** [comment]
Updated to split Features and Enhancements. Ordering in sections is reverse by issue ID (i.e. oldest at the bottom).

File: v1.4.0.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+- [FEATURE] Support CPU Pinning [#2305](https://github.com/harvester/harvester/issues/2305)
+- [GUI] [FEATURE] Support CPU Pinning [#6056](https://github.com/harvester/harvester/issues/6056)
+- [FEATURE] Restore vm backup to other harvester cluster [#2237](https://github.com/harvester/harvester/issues/2237)
+- [GUI] [FEATURE] Restore vm backup to other harvester cluster [#6371](https://github.com/harvester/harvester/issues/6371)
+- [FEATURE] Need to support RWX of Harvester CSI Driver [#1992](https://github.com/harvester/harvester/issues/1992)
+- [FEATURE] USB passthrough support [#1710](https://github.com/harvester/harvester/issues/1710)
+- [FEATURE] Support Bundle timeout enhancement  [#1646](https://github.com/harvester/harvester/issues/1646)
+- [FEATURE] Support setting hostname via DHCPv4 client [#1444](https://github.com/harvester/harvester/issues/1444)
+- [ENHANCEMENT] [GUI] Show warning message for Longhorn V2 unsupported actions [#6989](https://github.com/harvester/harvester/issues/6989)
+- [backport v1.4] [ENHANCEMENT][GUI] Move enable CPU Pinning checkbox position in template config page [#6937](https://github.com/harvester/harvester/issues/6937)
```
Comment (innobead) [inline v1.4.0.md:184]: This is the first release for 1.4. Why do we have a backport issue here?  This might be related to issue handling.

**Review by innobead — COMMENTED** [review]
In general, LGTM. Just a few comments.

File: v1.4.0.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+## Upgrades
+
+Harvester only allows upgrades from supported versions. For more information about upgrade paths and procedures, see [Upgrading Harvester](https://docs.harvesterhci.io/v1.4/upgrade/index).
+
+## Highlights
+
+### Technical Preview and Experimental Features
+
+#### Longhorn V2 Data Engine Support
```
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply v1.4.0.md:62]: We can if you think that's necessary. I based the list on Alex's slide deck for v1.4, which combines those two feature types into a single group. The heading used is "Technical Previews". How do you want to proceed? @innobead @bk201

**Comment by tserong** [comment]
I've just added all the bug fixes, but this list might want trimming (e.g. bugs that were found in v1.4 RCs but were fixed probably should not be listed)

File: v1.4.0.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+#### RWX Volumes for Guest Cluster Workloads
+
+Workloads running on guest clusters may require [read-write-many (RWX) volumes](https://docs.harvesterhci.io/v1.4/rancher/csi-driver#rwx-volumes-support) for backups and other purposes. Harvester v1.4 allows you to create RWX StorageClasses and associated volumes that can be attached to workloads through the Harvester CSI driver.
+
+To create RWX volumes, you must first create an RWX StorageClass on both the host Harvester cluster and the guest cluster, and an associated RWX PersistentVolumeClaim (PVC). The RWX PVC must be specified when creating pods for workloads that require RWX volumes.
+
+[View demo](https://youtu.be/5kHelpt1lUs)
+
+#### Virtual Machine Restoration in a New Harvester Cluster
```
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply v1.4.0.md:102]: We can. If we are to add the GitHub issue, I would like to change the presentation of external links. Here is the suggested structure:

> Highlight Title
> 
> Highlight description text
> 
> Demo: %link% | Doc: %link% | Issue: %link%

Should I start implementing this? @innobead @bk201

**Review by jillian-maroket — COMMENTED** [review]
@innobead I replied to your comments. cc: @bk201

File: v1.4.0.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (innobead) [reply v1.4.0.md:62]: We can simply add two sections. One is `Technical Preview Features` and the other is `Experimental Features`. 

Or at least add a suffix to each to present the type of it such as `- Technical Preview` or `- Experimental`.

P.S. People might ask for the definition of each type. We should create a doc to explain this later, out of scope here. (Simply said, experimental means alpha, and TP means beta`.

File: v1.4.0.md (hunk 1 above)
Comment (innobead) [reply v1.4.0.md:184]: Thanks for the context. Sounds good.

File: v1.4.0.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (innobead) [reply v1.4.0.md:102]: LGTM. The line may be long since all items are in line, but I know you'll find a better solution.

**Comment by tserong** [comment]
I've dropped a few bugs that I _know_ happened during v1.4 development and thus aren't relevant for the release notes. I've also removed a couple of backport tags and fixed some spacing.

File: v1.4.0.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply v1.4.0.md:62]: I have changed the heading. Per GitHub issues, this release includes 3 Experimental features and 0 Technical Preview features. Please confirm.

File: v1.4.0.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply v1.4.0.md:104]: I decided to hide the actual links. "Demo | Documentation | GitHub Issue" is much cleaner.

**Review by jillian-maroket — COMMENTED** [review]
@innobead Please check the updated heading and links.

**Review by innobead — APPROVED** [review]
LGTM. Thanks @jillian-maroket and @tserong

File: v1.4.0.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (innobead) [reply v1.4.0.md:62]: I think YES. @bk201

**Review by tserong — APPROVED** [review]
I believe we are complete now

**Review by bk201 — APPROVED** [review]

**Pull request merged**